tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +862,329 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Min<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 0);
+   EXPECT_EQ(hs["nominal"], 0);
+   EXPECT_EQ(hs["x:0"], -1);
+   EXPECT_EQ(hs["x:1"], 1);
 }
 
-TEST_P(RDFVary, DISABLED_VaryProfile1D)
+TEST_P(RDFVary, VaryProfiles)
 {
+   auto df = ROOT::RDataFrame(10)
+                .Define("x",
+                        [] {
+                           return ROOT::RVecI{1, 2, 3};
+                        })
+                .Vary(
+                   "x",
+                   [] {
+                      return ROOT::RVec<ROOT::RVecI>{{}, {4, 5}};
+                   },
+                   {}, 2);
+   auto h1 = df.Profile1D<ROOT::RVecI, ROOT::RVecI>({"name", "title", 100, 0, 100, 0, 100}, "x", "x");
+   auto h1s = VariationsFor(h1);
+   auto h2 =
+      df.Profile2D<ROOT::RVecI, ROOT::RVecI, ROOT::RVecI>({"name", "title", 100, 0, 100, 100, 0, 100}, "x", "x", "x");
+   auto h2s = VariationsFor(h2);
 
-}
+   EXPECT_DOUBLE_EQ(h1->GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2->GetMean(), 2.);
 
-TEST_P(RDFVary, DISABLED_VaryProfile2D)
-{
+   EXPECT_DOUBLE_EQ(h1s["nominal"].GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2s["nominal"].GetMean(), 2.);
 
+   EXPECT_DOUBLE_EQ(h1s["x:0"].GetMean(), 0.);
+   EXPECT_DOUBLE_EQ(h2s["x:0"].GetMean(), 0.);
+
+   EXPECT_DOUBLE_EQ(h1s["x:1"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(h2s["x:1"].GetMean(), 4.5);
 }
 
-TEST_P(RDFVary, DISABLED_VaryReduce)
+TEST(RDFVary, VaryReduce)
 {
-
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Reduce([](int x, int y) { return x + y; }, "x", 0);
+   auto hs = VariationsFor(h);
+   EXPECT_EQ(*h, 45);
+   EXPECT_EQ(hs["nominal"], 45);
+   EXPECT_EQ(hs["x:0"], 35);
+   EXPECT_EQ(hs["x:1"], 55);
 }
 
-TEST_P(RDFVary, DISABLED_VaryReport)
+TEST_P(RDFVary, VaryReport)
 {
-
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Filter([](int x) { return x > 5; }, {"x"}, "before")
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Filter([](int x) { return x > 7; }, {"x"}, "after")
+               .Report();
+   testing::internal::CaptureStdout();
+   h->Print();
+   std::string output = testing::internal::GetCapturedStdout();
+   auto expOut = "before    : pass=4          all=10         -- eff=40.00 % cumulative eff=40.00 %\n"
+                 "after     : pass=2          all=4          -- eff=50.00 % cumulative eff=20.00 %\n";
```

Comment:
No need to capture the output, you can check `h["before"]` and `h["after"]` programmatically, see https://root.cern/doc/master/classROOT_1_1RDF_1_1RCutFlowReport.html

Reply:
Got it - I am now testing with the getters from https://root.cern/doc/master/classROOT_1_1RDF_1_1TCutInfo.html